services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package ai.quantumics.api.service.impl;
+
+import ai.quantumics.api.exceptions.InvalidConnectionTypeException;
+import ai.quantumics.api.model.AWSDatasource;
+import ai.quantumics.api.model.Projects;
+import ai.quantumics.api.repo.AwsConnectionRepo;
+import ai.quantumics.api.req.AwsDatasourceRequest;
+import ai.quantumics.api.service.AwsConnectionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class AwsConnectionServiceImpl implements AwsConnectionService {
+
+    @Autowired
+    private AwsConnectionRepo awsConnectionRepo;
+
+    @Override
+    public AWSDatasource saveConnectionInfo(AwsDatasourceRequest awsDatasourceRequest, Projects project) {
+
+        String connectionType = awsDatasourceRequest.getConnectionType();//access Type
+
+        if ("IAM".equals(connectionType)) {
+            AWSDatasource awsDatasource = awsDatasourceMapper(awsDatasourceRequest, project);
+
+            return awsConnectionRepo.saveAndFlush(awsDatasource);
```

Comment:
most of the basic validations are missing

Reply:
Done

Reply:
what is done?